### PR TITLE
Refactored a bit for further development

### PR DESCRIPTION
**Pull request by sawickil**
I know it looks like a huge disaster, but let's try to make this code more structured. However, I believe there are a few things left;)

### CODE REVIEW

File: src/main/java/com/github/yu55/gog/core/GrepController.java [hunk 1]
```diff
@@ -6,30 +9,29 @@
...
-    public Controller(GrepService grepService, Repositories repositories) {
+    public GrepController(GrepService grepService) {
         this.grepService = grepService;
-        this.repositories = repositories;
     }
 
     @RequestMapping(value = "/repositories")
     public List<String> repositories() {
-        return repositories.getDirectories().stream().map(d -> d.getName()).collect(Collectors.toList());
+        return grepService.getAvailavleRepositories();
```
Comment (yu55) [inline src/main/java/com/github/yu55/gog/core/GrepController.java]: Availavle -> Available

File: src/main/java/com/github/yu55/gog/core/GrepController.java [hunk 2]
```diff
@@ -6,30 +9,29 @@
...
     }
 
     @RequestMapping(value = "/grep", method = RequestMethod.POST)
     public List<String> grep(@RequestBody GrepRequest grepRequest) {
-        return grepService.grep(grepRequest);
+        return grepService.grep(grepRequest).getResponseLines()
+                .stream()
+                .map(responseLine -> responseLine.getLine())
```
Comment (yu55) [inline src/main/java/com/github/yu55/gog/core/GrepController.java]: Replace lambda expression with ResponseLine::getLine

File: src/main/java/com/github/yu55/gog/core/model/GrepResponseLine.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.github.yu55.gog.core.model;
+
+/**
+ * Created by lsaw on 10/13/15.
```
Comment (yu55) [inline src/main/java/com/github/yu55/gog/core/model/GrepResponseLine.java]: Please remove this JavaDoc

File: src/main/java/com/github/yu55/gog/core/model/GrepResponse.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package com.github.yu55.gog.core.model;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class GrepResponse {
+
+    public static final int RESPONSE_LINES_LIMIT = 1001;
```
Comment (yu55) [inline src/main/java/com/github/yu55/gog/core/model/GrepResponse.java]: Consider changing scope to private.

File: src/main/java/com/github/yu55/gog/core/GrepService.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    @Autowired
+    public GrepService(GitGrepHandler gitGrepHandler) {
+        this.gitGrepHandler = gitGrepHandler;
+    }
+
+    public GrepResponse grep(GrepRequest wanted) {
+        return gitGrepHandler.grep(wanted);
+    }
+
+    public List<String> getAvailavleRepositories() {
```
Comment (yu55) [inline src/main/java/com/github/yu55/gog/core/GrepService.java]: Availavle -> Available

File: src/main/java/com/github/yu55/gog/core/GrepService.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.stereotype.Component;
+
+import com.github.yu55.gog.core.model.GrepRequest;
+import com.github.yu55.gog.core.model.GrepResponse;
+import com.github.yu55.gog.handler.git.GitGrepHandler;
+
+@Component
+public class GrepService {
+
+    private final GitGrepHandler gitGrepHandler;
```
Comment (yu55) [inline src/main/java/com/github/yu55/gog/core/GrepService.java]: So, for now GrepService will have hardcoded "git grepping"?

File: src/main/java/com/github/yu55/gog/core/model/GrepRequest.java [hunk 7]
```diff
@@ -1,15 +1,21 @@
-package com.github.yu55.gog;
+package com.github.yu55.gog.core.model;
 
 import java.util.List;
 
 public class GrepRequest {
+
     private String wanted;
+
     private List<String> repositories;
```
Comment (yu55) [inline src/main/java/com/github/yu55/gog/core/model/GrepRequest.java]: Consider more meaningful name: repositories -> repositoriesToSearch

File: src/main/java/com/github/yu55/gog/handler/git/GitRepository.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package com.github.yu55.gog.handler.git;
+
+import java.io.File;
+
+public class GitRepository {
```
Comment (yu55) [inline src/main/java/com/github/yu55/gog/handler/git/GitRepository.java]: No abstraction for repository?

Comment (yu55) [inline src/main/java/com/github/yu55/gog/handler/git/GitRepository.java]: Overall I like the direction of changes.
Please check comments.

File: src/main/java/com/github/yu55/gog/core/model/GrepResponse.java (hunk 4 above)
Comment (sawickil) [inline src/main/java/com/github/yu55/gog/core/model/GrepResponse.java]: this is used outside this class

**Pull request merged**